dash-renderer/src/observers/requestedCallbacks.ts

```diff
@@ -79,25 +87,35 @@ const observer: IStoreObserverDefinition<IStoreState> = {
         */
 
         /*
-            Extract all but the first callback from each IOS-key group
-            these callbacks are duplicates.
+            Group callbacks by identifier and partition based on whether there are duplicates or not.
         */
-        const rDuplicates = flatten(map(
-            group => group.slice(0, -1),
-            values(
-                groupBy<ICallback>(
-                    getUniqueIdentifier,
-                    requested
-                )
+        const [rWithoutDuplicates, rWithDuplicates] = partition(rdg => rdg.length === 1, values(
+            groupBy<ICallback>(
+                getUniqueIdentifier,
+                requested
             )
         ));
 
+        /*
+            Flatten all duplicated callbacks into a list for removal
+        */
+        const rDuplicates = flatten(rWithDuplicates);
+
+        /*
+            Merge duplicate groups into a single callback - merge by giving priority to newer callbacks
+        */
+        const rMergedDuplicates = map(group => assoc(
+            'changedPropIds',
+            mergeAll(pluck('changedPropIds', group)),
```

Comment:
In order to ensure `DIRECT` doesn't turn into `INDIRECT` we need something like:

https://github.com/plotly/dash/blob/d9a8d9c2686fddd2a9a6884f9ed34bc44f1c767d/dash-renderer/src/actions/dependencies_ts.ts#L25

Reply:
@alexcjohnson Thanks for the heads up. Also distinguishing the initial callback from subsequent callbacks https://github.com/plotly/dash/pull/1385/commits/8b6ff26341acbd39f0f3d0fa1066bc89d2255eec#diff-178b2bd05a773877fd1b117eef8b1e8cR90